### PR TITLE
Draft: experimental new .select and dynamic fn support

### PR DESCRIPTION
**Pull request by adamvy**

### CODE REVIEW

File: src/foam/u2/Element2.js [hunk 1]
```diff
@@ -1223,7 +1247,51 @@ foam.CLASS({
...
+        // every time the dynamic function runs
+        // remove the previous contents
+        // and run the function so that
+        // the new contents is inserted where the marker is
+
+
+        // nested dynamic functions
+        // a dyamic fucntion might itself call anotehr dynamic function
+        // in which case, when we clean up our contents we should
+        // remove any contents that hte function generated
```
Comment (kgrgreer) [inline src/foam/u2/Element2.js:1261]: s/hte/the

File: src/foam/u2/Element2.js [hunk 2]
```diff
@@ -1376,11 +1438,44 @@ foam.CLASS({
      * the DAO
      */
     function select(dao, f, update, opt_comparator) {
-      this.add(foam.u2.DAOSelectNode.create({
-        self: this,
-        dao: dao,
-        code: f
+      let ref = foam.u2.Text.create({ text: '' });
```
Comment (kgrgreer) [inline src/foam/u2/Element2.js:1441]: We don't need DAOSelectNode anymore?

File: src/foam/u2/Element2.js [hunk 3]
```diff
@@ -1223,7 +1247,51 @@ foam.CLASS({
         c = c.toE(null, this.__subSubContext__);
       }
       if ( foam.core.DynamicFunction.isInstance(c) ) {
-        this.addChild_(foam.u2.FunctionNode.create({fn: c, parentNode: this}, this), this);
+        // when adding a "dynamic function"
```
Comment (kgrgreer) [inline src/foam/u2/Element2.js:1250]: We don't need FunctionNode anymore?

**Comment by adamvy** [comment]
Superceded by https://github.com/kgrgreer/foam3/commit/24945bf9b4c7b16b7155595e453e6452e0f48ca4 which is a cleaner impelmentation of this idea :)

**Pull request closed without merging**